sympy/core/basic.py

```diff
@@ -1332,105 +1338,106 @@ def replace(self, query, value, map=False, simultaneous=True, exact=None):
             argument(s). ``func`` must be written to handle the number
             of arguments of ``type``.
 
-            >>> f.replace(sin, lambda arg: sin(2*arg))
-            log(sin(2*x)) + tan(sin(2*x**2))
-            >>> (x*y).replace(Mul, lambda *args: sin(2*Mul(*args)))
-            sin(2*x*y)
+        >>> f.replace(sin, lambda arg: sin(2*arg))
+        log(sin(2*x)) + tan(sin(2*x**2))
+        >>> (x*y).replace(Mul, lambda *args: sin(2*Mul(*args)))
+        sin(2*x*y)
 
         2.1. pattern -> expr
             obj.replace(pattern(wild), expr(wild))
 
             Replace subexpressions matching ``pattern`` with the expression
             written in terms of the Wild symbols in ``pattern``.
 
-            >>> a, b = map(Wild, 'ab')
```

Comment:
The indentation here follows the logic of the section's indentation: they are tests under section 2.1. Generally, the author chooses how to indent the examples relative to the discussion. Does it print differently when it is rendered?

Reply:
Yes, it prints differently.
The code block prints in line with the previous code block, an indent behind the text.

Reply:
I'm not sure how I missed @sylee957 comments. I think I am agreeing by saying that how it looked previously was fine. I like how the code is indented in correspondence with the text with which it appears.

Reply:
ok then, I'l revert the changes for `replace`

Reply:
I think the 4-spacing changes for `Basic`, `_subs`, `replace` can be reverted.
Then it would not be blocking the merge.

Reply:
> I think the 4-spacing changes for `Basic`, `_subs`, `replace` can be reverted.
> Then it would not be blocking the merge.

Is there a merge conflict?

do you want me to revert the changes in `Basic` and `_subs` too?

Reply:
I just mean by the thing people would have consensus on.
I see these 4-spacing indentations are having no problem, and the previous intent was even better for the visual right now, because it wraps the examples under the subsections.

I see `Basic` and `_subs` are also having 4-character indentation for intent like above.
So please revert them.